tests/Fixer/Operator/NoUselessConcatOperatorFixerTest.php

```diff
@@ -136,6 +136,33 @@ public static function provideFixCases(): iterable
             ',
         ];
 
+        yield 'do not fix if variables would be broken' => [
+            '<?php
+                echo "abc $d" . "e $f";
+                echo "abc $d" . "e";
+                echo "abc $d" . \'e\';
+                echo "abc $d" . "😃"; // with emoji
+                echo "私の名前は$name" . "です"; // multibyte characters (Japanese)
+            ',
+        ];
+
+        yield 'fix if variables would not be broken' => [
+            '<?php
+                echo "$a bcde";
+                echo "abc ${d}e";
+                echo "abc $d-efg";
+                echo "abc $d-$e";
+                echo "abc $d[$e]";
```

Comment:
This particular line should not be changed, `"abc $d[$e]"` is not the same as `"abc $d" . "[$e]"`.

Note that this is also true with `"abc {$d{$e}}"` which is the same as `"abc $d[$e]"` and allowed on PHP 7.4 (invalid on PHP 8.0+).

Reply:
I was careless... That's right.

And furthermore, `"abc $d" . "->e"` should also not be fixed to `"abc $d->e"` 😖